src/main/java/org/jvnet/hudson/test/InboundAgentRule.java

```diff
@@ -440,7 +460,15 @@ public void start(AgentArguments agentArguments, Options options) throws Excepti
         pb.environment().put("INBOUND_AGENT_RULE_NAME", options.getName());
         LOGGER.info(() -> "Running: " + pb.command());
         Process proc = pb.start();
-        procs.put(options.getName(), proc);
+        procs.compute(options.getName(), (k, v) -> {
+            if (v != null) {
+                // Duplicate agent name, but this can be a valid test case.
+                var result = new ArrayList<>(v);
+                result.add(proc);
+                return result;
+            }
+            return List.of(proc);
+        });
```

Comment:
BTW there is some one-liner to do this using `Map.merge` or something like that.